framework/src/modules/chain/bft/consensus_manager.js

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const assert = require('assert');
+const debug = require('debug')('lisk:bft:consensus_manager');
+const { HeadersList } = require('./headers_list');
+const { validateBlockHeader } = require('./utils');
+const {
+	BFTChainDisjointError,
+	BFTLowerChainBranchError,
+	BFTForkChoiceRuleError,
+	BFTInvalidAttributeError,
+} = require('./errors');
+
+/**
+ * @typedef {Object} BlockHeader
+ * @property {string} blockId
+ * @property {int} height
+ * @property {int} maxHeightPreviouslyForged
+ * @property {int} prevotedConfirmedUptoHeight
+ * @property {int} activeSinceRound
+ * @property {string} delegatePublicKey
+ */
+
+class ConsensusManager {
```

Comment:
I also felt `ConsensusManager` is kind of misleading, too. Although it does reflect what it's doing.
Im not sure if ConsensusManager is good name to expose to the external either.

Maybe, instead of swapping, we can come up with better name for this class?

Reply:
For me, it feels that as it is being managed by `BFT.js`, hence `Manager` isn't a name that reflects its purspose (?). 

> Maybe, instead of swapping, we can come up with better name for this class?
Agreed

Reply:
We discussed for this name convention for a long time and then used it. I originally named it as `FinalityManager` and then on request of @SargeKhan changed it to `ConsensusManager`. 

To me `FinalityManager` and `ConsensusManager` names both relates to the business logic it includes. `BFT` itself is very broad term which covers every thing so better suited for integration logic. This file only have particular scope to manage one particular aspect in the BFT.

Reply:
In my opinion the name of `Byzantine Fault Tolerant(nce)` for an integration class it's not self-describing enough. For me it lacks a `verb` after it to expose what is its purpose, that is why I proposed adding the `Manager` or something similar word to the integration logic.

Reply:
@limiaspasdaniel This logic applies to every module we have. `blocks`, `dpos`